src/main/java/de/retest/web/selenium/AutocheckingNavigationWrapper.java

```diff
@@ -0,0 +1,48 @@
+package de.retest.web.selenium;
+
+import java.net.URL;
+
+import org.openqa.selenium.WebDriver.Navigation;
+
+public class AutocheckingNavigationWrapper implements Navigation {
+
+	private final Navigation delegate;
+	private final AutocheckingRecheckDriver autocheckingDriver;
+
+	public AutocheckingNavigationWrapper( final Navigation delegate,
+			final AutocheckingRecheckDriver autocheckingDriver ) {
+		this.delegate = delegate;
+		this.autocheckingDriver = autocheckingDriver;
+	}
+
+	@Override
+	public void back() {
+		delegate.back();
+		autocheckingDriver.check( "navigate-back" );
+	}
+
+	@Override
+	public void forward() {
+		delegate.forward();
+		autocheckingDriver.check( "navigate-backforward" );
+	}
+
+	@Override
+	public void to( final String url ) {
+		delegate.to( url );
+		autocheckingDriver.check( "navigate-to" );
+	}
+
+	@Override
+	public void to( final URL url ) {
+		delegate.to( url );
```

Comment:
If you want to, you can reuse the method above:

```java
@Override
public void to( final URL url ) {
	to( url.toString() );
}
```

Reply:
Hmm. Might make a difference for the delegate, so I'd rather only call that...